static/js/narrow.js

```diff
@@ -535,6 +551,10 @@ function pick_empty_narrow_banner() {
             // You have no private messages.
             return $("#empty_narrow_all_private_message");
         }
+    } else if ((first_operator === "stream") && !stream_data.is_subscribed(first_operand)
+                                             && sub.invite_only) {
+        // You are narrowed to a invite only stream to which you aren't subscribed
+        return $("#non_existing_stream_message");
```

Comment:
I'm not sure we need this check -- is the stream is `invite_only`, the user won't have any data on the stream anyway because of how the backend works, right?  Or am I missing something?